Sources/Filters+Strings+Lettercase.swift

```diff
@@ -0,0 +1,157 @@
+    }
+
+    /// Converts camelCase to snake_case. Takes an optional Bool argument for making the string lower case,
+    /// which defaults to true
+    ///
+    /// - Parameters:
+    ///   - value: the value to be processed
+    ///   - arguments: the arguments to the function; expecting zero or one boolean argument
+    /// - Returns: the snake case string
+    /// - Throws: FilterError.invalidInputType if the value parameter isn't a string
+    static func camelToSnakeCase(_ value: Any?, arguments: [Any?]) throws -> Any? {
+        let toLower = try Filters.parseBool(from: arguments, required: false) ?? true
+        guard let string = value as? String else { throw Filters.Error.invalidInputType }
+
+        let snakeCase = try snakecase(string)
+        if toLower {
+            return snakeCase.lowercased()
+        }
+        return snakeCase
+    }
+
+    /// Converts snake_case to camelCase, stripping prefix underscores if needed
+    ///
+    /// - Parameters:
+    ///   - string: the value to be processed
+    ///   - stripLeading: if false, will preserve leading underscores
+    /// - Returns: the camel case string
+    static func snakeToCamelCase(_ string: String, stripLeading: Bool) throws -> String {
+        let unprefixed: String
+        if try containsAnyLowercasedChar(string) {
+            let comps = string.components(separatedBy: "_")
+            unprefixed = comps.map { titlecase($0) }.joined(separator: "")
+        } else {
+            let comps = try snakecase(string).components(separatedBy: "_")
+            unprefixed = comps.map { $0.capitalized }.joined(separator: "")
+        }
+
+        // only if passed true, strip the prefix underscores
+        var prefixUnderscores = ""
+        var result: String { return prefixUnderscores + unprefixed }
+        if stripLeading {
+            return result
+        }
+        for scalar in string.unicodeScalars {
+            guard scalar == "_" else { break }
+            prefixUnderscores += "_"
+        }
+        return result
+    }
+
+    // MARK: - Private
+
+    private static func containsAnyLowercasedChar(_ string: String) throws -> Bool {
+        let lowercaseCharRegex = try NSRegularExpression(pattern: "[a-z]", options: .dotMatchesLineSeparators)
+        let fullRange = NSRange(location: 0, length: string.unicodeScalars.count)
+        return lowercaseCharRegex.firstMatch(in: string, options: .reportCompletion, range: fullRange) != nil
+    }
+
+    /// This returns the string with its first parameter uppercased.
+    /// - note: This is quite similar to `capitalise` except that this filter doesn't
+    ///          lowercase the rest of the string but keeps it untouched.
+    ///
+    /// - Parameter string: The string to titleCase
+    /// - Returns: The string with its first character uppercased, and the rest of the string unchanged.
+    private static func titlecase(_ string: String) -> String {
```

Comment:
I wonder why that `titlecase` name came back up here, did you copy/paste the code from _before_ the last commits which changed that name when you started working on that split? Or did we miss something when doing our latest merges and let that slip?